src/consultations/consultations.controller.ts

```diff
@@ -449,4 +450,94 @@ export class ConsultationsController {
       HttpStatus.NOT_FOUND,
     )
   }
+
+  @JwtAuth()
+  @RequiredPermission(Permissions.JoinConsultation)
+  @Get(':id/ics')
+  async getCalendar(
+    @Param('id', ParseIntPipe) id: number,
+    @CurrentUser() user: UserEntity,
+    @Res() res: Response,
+  ): Promise<StreamableFile> {
```

Comment:
the reqiredPermission line is not necessary. `Permissions.JoinConsultation` would prevent owners and presenters from accessing this endpoint (since they aren't allowed to join their own consultation) You don't have to worry about permissions here, findOne will throw a 404 if the user doesn't have read rights, and otherwise they are allowed to export the file.

And as Dani said, `@Res({ passthrough: true }) res: Response` is needed